test/support/otlp_endpoint_test.rb

```diff
@@ -0,0 +1,250 @@
+# frozen_string_literal: true
+
+# Copyright (c) 2023 SolarWinds, LLC.
+# All rights reserved.
+
+require 'minitest_helper'
+require './lib/solarwinds_apm/config'
+require './lib/solarwinds_apm/support/otlp_endpoint'
+require './lib/solarwinds_apm/support/service_key_checker'
+
+# BUNDLE_GEMFILE=gemfiles/unit.gemfile bundle exec ruby -I test test/support/otlp_endpoint_test.rb
+describe 'OTLP Endpoint Test' do
+  before do
+    @original_env = ENV.to_h.dup
+    ENV.clear
+  end
+
+  after do
+    ENV.replace(@original_env)
+  end
+
+  # 2,5,8 (failed to auth token, not test here)
+  describe 'config_token' do
+    # 1
+    it 'wrong formatted SW_APM_SERVICE_KEY OTEL PROTO' do
+      ENV['SW_APM_SERVICE_KEY'] = nil
+      ENV['OTEL_EXPORTER_OTLP_ENDPOINT'] = 'https://otel.collector.na-01.cloud.solarwinds.com'
+      endpoint = SolarWindsAPM::OTLPEndPoint.new
+      endpoint.config_token
+
+      assert_nil(endpoint.instance_variable_get(:@token))
+      assert_nil(endpoint.instance_variable_get(:@service_name))
+      _(endpoint.instance_variable_get(:@agent_enable)).must_equal false
+    end
+
+    # 3
+    it 'correct formatted SW_APM_SERVICE_KEY, successful auth token OTEL PROTO' do
+      ENV['SW_APM_SERVICE_KEY'] = '0123456789abcde0123456789abcde0123456789abcde0123456789abcde1234:my_service'
+      ENV['OTEL_EXPORTER_OTLP_ENDPOINT'] = 'https://otel.collector.na-01.cloud.solarwinds.com'
+      endpoint = SolarWindsAPM::OTLPEndPoint.new
+      endpoint.config_token
+
+      _(endpoint.instance_variable_get(:@token)).must_equal '0123456789abcde0123456789abcde0123456789abcde0123456789abcde1234'
+      _(endpoint.instance_variable_get(:@service_name)).must_equal 'my_service'
+      _(endpoint.instance_variable_get(:@agent_enable)).must_equal true
+      _(ENV.fetch('OTEL_EXPORTER_OTLP_HEADERS', nil)).must_equal 'authorization=Bearer 0123456789abcde0123456789abcde0123456789abcde0123456789abcde1234'
```

Comment:
Can we add assertions here and in the rest of the tests on what the resolved OTLP endpoint is?

Reply:
For current test structure, if testing `config_token` (e.g. `describe 'config_token'`; `endpoint.config_token`), it only configure the token value, so all endpoint will be nil unless I explicitly set env variable e.g. `OTEL_EXPORTER_OTLP_ENDPOINT` to some value. But in real world case, if noting provided, then it will use our default endpoint

Reply:
Ah i see. For the auth token behaviour this is what i'd expect:

OTLP exporter auth header is set from SW service key token portion, only if:
* the OTLP exporter header config is not already set, AND
* the OTLP exporter endpoint is SWO otel.collector.*.solarwinds.com

I think the test cases in this `config_token` suite can be simplified to focus on the above behaviour since the same applies to both inside and outside lambda.

Reply:
I kind want to keep the existing test case as it tries to reflect the test case for confluence page.
I also want to make sure that we don't need to set SW service key from deriving from HEADERS (e.g. HEADERS set but no SW service key).